src/libpmemblk/blk.c

```diff
@@ -537,6 +538,14 @@ pmemblk_create(const char *path, size_t bsize, size_t poolsize,
 
 	PMEMblkpool *pbp;
 	pbp = pmemblk_map_common(fd, poolsize, bsize, 0, 1, created);
+	if (pbp && created) {
+		if (chmod(path, mode)) {
```

Comment:
You have the fd (used immediately above), so I would change these calls to fchmod().

Should the file be unlinked if the chmod fails?  Any reason to leave it around?